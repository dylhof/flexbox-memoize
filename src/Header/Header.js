import React, { Component } from 'react';
import Instructions from '../Instructions/Instructions.js';

class Header extends Component {
  constructor() {
    super();
    this.state = {
      showInstructions: false
    }
  }
  
  toggleShowInstructions = () => {
    this.setState({ showInstructions: !this.state.showInstructions })
  }

  render() {
    return (
      <header className="App-header">
        <h1>FlexboxQuiz</h1>
        <i className="far fa-question-circle" onClick={this.toggleShowInstructions}></i>
        <Instructions 
          showInstructions={this.state.showInstructions}
          toggleShowInstructions={this.toggleShowInstructions}/>
      </header>
    );
  }
}

export default Header;