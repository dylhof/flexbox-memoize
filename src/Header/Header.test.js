import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Header />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should toggle showInstructions', () => {
    expect(wrapper.state('showInstructions')).toEqual(false);
    wrapper.instance().toggleShowInstructions();
    expect(wrapper.state('showInstructions')).toEqual(true);
    wrapper.instance().toggleShowInstructions();
    expect(wrapper.state('showInstructions')).toEqual(false);
  })
})