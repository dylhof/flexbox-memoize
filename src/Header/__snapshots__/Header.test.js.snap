// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header
  className="App-header"
>
  <h1>
    FlexboxQuiz
  </h1>
  <i
    className="far fa-question-circle"
    onClick={[Function]}
  />
  <Instructions
    showInstructions={false}
    toggleShowInstructions={[Function]}
  />
</header>
`;
